books/RayTracingTheNextWeek.html

```diff
@@ -1045,41 +1068,46 @@
                 auto k = static_cast<int>(floor(p.z()));
                 return ranfloat[perm_x[i] ^ perm_y[j] ^ perm_z[k]];
             }
-            static double *ranfloat;
-            static int *perm_x;
-            static int *perm_y;
-            static int *perm_z;
+
+        public:
+            static const int point_count = 256;
+            static double* ranfloat;
+            static int* perm_x;
+            static int* perm_y;
+            static int* perm_z;
     };
 
     static double* perlin_generate() {
-        auto p = new double[256];
-        for (int i = 0; i < 256; ++i)
+        auto p = new double[perlin::point_count];
```

Comment:
Are we not moving `perlin` over to `vector`s and `shared_ptr`s?

Reply:
Good catch. Will fix.

Reply:
No need. These basis values are never shared out, but are only used to compute Perlin values. In this case, a static array is sufficient (and simple), and has the lifetime we need (across many uses & frames if need be). It just needs to be initialize once at the beginning of the program.

Still, this caught a fragment of old code, which allocated the buffer. The new code just declares a static array.

Reply:
Will update.